internal/resource/procfs_reader.go

```diff
@@ -0,0 +1,126 @@
+// SPDX-FileCopyrightText: 2025 The Kepler Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package resource
+
+import (
+	"fmt"
+
+	"github.com/prometheus/procfs"
+)
+
+// cGroup holds only required cgroup info about the process
+type cGroup struct {
+	Path string // used to detect if a process is running in a container
+}
+
+// procInfo is an interface that wraps the necessary methods from procfs.Proc to be used by the resource service
+type procInfo interface {
+	PID() int
+	Comm() (string, error)
+	Executable() (string, error)
+	Cgroups() ([]cGroup, error)
+	Environ() ([]string, error)
+	CmdLine() ([]string, error)
+	CPUTime() (float64, error)
+}
+
+// procWrapper implements ProcInfo by wrapping procfs.Proc. This is needed because the procfs.Proc
+// does not implement PID() as a method
+type procWrapper struct {
+	proc procfs.Proc
+}
+
+var _ procInfo = (*procWrapper)(nil)
+
+func (p *procWrapper) PID() int {
+	return p.proc.PID
+}
+
+func (p *procWrapper) Comm() (string, error) {
+	return p.proc.Comm()
+}
+
+func (p *procWrapper) Executable() (string, error) {
+	return p.proc.Executable()
+}
+
+func (p *procWrapper) Cgroups() ([]cGroup, error) {
+	cgroupsData, err := p.proc.Cgroups()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get process cgroups: %w", err)
+	}
+
+	cgroups := make([]cGroup, len(cgroupsData))
+	for i, cg := range cgroupsData {
+		cgroups[i] = cGroup{
+			Path: cg.Path,
+		}
+	}
+	return cgroups, nil
+}
+
+func (p *procWrapper) Environ() ([]string, error) {
+	return p.proc.Environ()
+}
+
+func (p *procWrapper) CmdLine() ([]string, error) {
+	return p.proc.CmdLine()
+}
+
+// userHZ is the number of clock ticks per second
+// hardcoded just like in procfs
+const userHZ = 100
```

Comment:
```
/*
#include <unistd.h>
*/
import "C"

userHZ = int64(C.sysconf(C._SC_CLK_TCK))

```

Reply:
Just like in node-exporter, lets only hardcode it and not have CGO unless really needed